test/ibmq/test_ibmq_jobmanager.py

```diff
@@ -303,6 +303,160 @@ def test_job_tags(self, provider):
+                                 .format(job_id, job.tags(), tags_after_appending))
+
+                # Ensure the job set long id is in the new tags.
+                self.assertIn(job_set._id_long, job.tags(),
+                              'The job {} with tags {} does not contain the job set '
+                              'long id "{}".'.format(job_id, job.tags(), job_set._id_long))
+
+                # Ensure the job is retrieved with the new tags it is associated with.
+                retrieved_jobs = provider.backends.jobs(job_tags=tags_after_appending)
+                retrieved_job_ids = {job.job_id() for job in retrieved_jobs}
+                self.assertIn(job_id, retrieved_job_ids)
+
+    @requires_provider
+    def test_job_tags_modify_overwrite(self, provider):
+        """Test modifying tags from a job, of a job set, overwriting its existing tags."""
+        backend = provider.get_backend('ibmq_qasm_simulator')
+
+        initial_job_tags = [uuid.uuid4().hex]
+        job_set = self._jm.run([self._qc] * 2, backend=backend,
+                               max_experiments_per_job=1, job_tags=initial_job_tags)
+
+        # Wait for jobs to be submitted.
+        while JobStatus.INITIALIZING in job_set.statuses():
+            time.sleep(1)
+        # TODO No need to wait for job to run once api is fixed
+        while any(status not in JOB_FINAL_STATES + (JobStatus.RUNNING,)
+                  for status in job_set.statuses()):
+            time.sleep(0.5)
+
+        # Only test overwriting the tags of one job, from the job set.
+        job = job_set.jobs()[0]
+        job_id = job.job_id()
+
+        timestamp = str(time.time()).replace('.', '')
+        new_tags_to_overwrite = [
+            [],  # empty tags.
+            ['{}_new_tag_{}'.format(timestamp, i) for i in range(2)]  # non-empty unique tags.
+        ]
+        # Overwrite the tags for a job in the job set.
+        for new_tags in new_tags_to_overwrite:
+            new_tags_with_id_long = new_tags + [job_set._id_long]
+
+            with self.subTest(new_tags=new_tags):
+                update_successful = job.modify_tags(new_tags, overwrite=True)
+                self.assertTrue(update_successful,
+                                'Updating the tags for job {} was unsuccessful.'
+                                'The tags are {}, but they should be {}.'
+                                .format(job_id, job.tags(), new_tags_with_id_long))
+
+                # Ensure the job's tags match the new tags and contain the job set long id.
+                self.assertEqual(set(job.tags()), set(new_tags_with_id_long),
+                                 'The tags for job {} are {}, but they should be {}'
+                                 .format(job_id, job.tags(), new_tags_with_id_long))
+
+                # Ensure the job set long id is in the new tags.
+                self.assertIn(job_set._id_long, job.tags(),
```

Comment:
This seems to be a duplicate check as above. If `job.tags()` equals to `new_tags_with_id_long` (which has `_id_long`), then `_id_long` must be in `job.tags()`.